test/tests/functional/pbs_qmgr.py

```diff
@@ -160,24 +160,42 @@ def test_multi_attributes(self):
             self.assertTrue(msg in e.msg[0])
         self.server.expect(NODE, 'queue', op=UNSET, id=self.mom.shortname)
 
-    def test_string_quoting(self):
+    def set_and_test_comment(self, comment):
         """
-        Test to verify that if a string attribute has spaces, the value
-        is quoted when printed
+        Set the node's comment, then print it and re-import it
         """
-        a = {'default_qsub_arguments': '-V -lplace=scatter'}
-        self.server.manager(MGR_CMD_SET, SERVER, a)
-        qmgr_cmd = \
-            os.path.join(self.server.pbs_conf['PBS_EXEC'], 'bin', 'qmgr') + \
-            " -c '%s'"
-        qmgr_cmd_print = qmgr_cmd % ('p s default_qsub_arguments')
+        a = {'comment': comment}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.hostname)
+        qmgr_path = \
+            os.path.join(self.server.pbs_conf['PBS_EXEC'], 'bin', 'qmgr')
+        qmgr_cmd_print = qmgr_path + \
+            (' -c "p n %s comment"' % self.mom.hostname)
         ret = self.du.run_cmd(self.server.hostname,
                               cmd=qmgr_cmd_print, as_script=True)
         self.assertEqual(ret['rc'], 0)
+        fn = self.du.create_temp_file()
         for line in ret['out']:
             if '#' in line:
                 continue
-            qmgr_cmd_check = qmgr_cmd % (line)
-            ret_s = self.du.run_cmd(
-                self.server.hostname, cmd=qmgr_cmd_check, as_script=True)
+            if 'create node' in line:
+                continue
+            with open(fn, 'w') as f:
```

Comment:
Shouldn't the mode be 'w+' as it has to be read by qmgr?

Reply:
`w+` open the file for updating, if it's already created. Otherwise it will create a new file.
`w` will truncate the file (which is what we want to do, since we only create one file and write to it many times.